x-pack/plugins/fleet/server/errors/handlers.ts

```diff
@@ -47,45 +54,72 @@ interface IngestErrorHandlerParams {
 }
 // unsure if this is correct. would prefer to use something "official"
 // this type is based on BadRequest values observed while debugging https://github.com/elastic/kibana/issues/75862
-
 const getHTTPResponseCode = (error: FleetError): number => {
-  if (error instanceof RegistryResponseError) {
-    // 4xx/5xx's from EPR
-    return 500;
+  // Bad Request
+  if (error instanceof PackageFailedVerificationError) {
+    return 400;
   }
-  if (error instanceof RegistryConnectionError || error instanceof RegistryError) {
-    // Connection errors (ie. RegistryConnectionError) / fallback  (RegistryError) from EPR
-    return 502; // Bad Gateway
+  if (error instanceof PackageOutdatedError) {
+    return 400;
   }
-  if (error instanceof PackageNotFoundError || error instanceof PackagePolicyNotFoundError) {
-    return 404; // Not Found
+  if (error instanceof PackageInvalidArchiveError) {
+    return 400;
   }
-  if (error instanceof AgentPolicyNameExistsError) {
-    return 409; // Conflict
+  if (error instanceof PackageRemovalError) {
+    return 400;
   }
-  if (error instanceof PackageUnsupportedMediaTypeError) {
-    return 415; // Unsupported Media Type
+  if (error instanceof KibanaSOReferenceError) {
+    return 400;
   }
-  if (error instanceof PackageFailedVerificationError) {
-    return 400; // Bad Request
+  // Unauthorized
+  if (error instanceof FleetUnauthorizedError) {
+    return 403;
```

Comment:
Should this be 401? Or this is independent of the authorization state?

Reply:
I think 403 is correct here as the user is authenticated but do not have all the permissions to perform a package install

Reply:
Then yes, 403 looks fine :+1: